### PR TITLE
GHA: Run unit tests in CI

### PR DESCRIPTION
**Pull request by limbonaut**
Run unit tests in CI using Windows and Linux runners. This will help us detect compilation and runtime issues early on.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: ⚙️ CI
 on:
   push:
     branches:
-      - "main"
+      # - "main"
```
Comment (bruno-garcia) [inline .github/workflows/ci.yml]:
```suggestion
      - "main"
```

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        include:
+          - runner: windows-latest
+            godot-suffix: win64.exe
+          - runner: ubuntu-latest
+            godot-suffix: linux.x86_64
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          submodules: false # don't initialize submodules automatically
```
Comment (bruno-garcia) [inline .github/workflows/unit_tests.yml:22]: Why the additional step instead of this?

Comment (limbonaut) [reply .github/workflows/unit_tests.yml:22]: Can't specify which submodules to check out... We don't need `sentry-native` and such, which also packs its own submodules on top. If you prefer simpler, I can remove that step.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (limbonaut) [reply .github/workflows/ci.yml]: That was [temporary](https://github.com/getsentry/sentry-godot/pull/37#issuecomment-2520008953).
Dropped.

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (bruno-garcia) [reply .github/workflows/unit_tests.yml:22]: All good

**Comment by limbonaut** [comment]
@bitsandfoxes @bruno-garcia
Good to merge? It's still waiting for approval.

**Review by bitsandfoxes — APPROVED** [review]
Awesome stuff!

File: .github/workflows/unit_tests.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          scons project/addons/gdUnit4
+          chmod +x project/addons/sentrysdk/bin/crashpad_handler
+          echo "--- Rebuilding import cache.."
+          ${GODOT_BIN} --headless --editor --path project/ --quit-after 2000 || true
+          echo "--- Finished rebuilding import cache."
+
+      - name: Run tests
+        shell: bash
+        timeout-minutes: 5
+        run: |
```
Comment (bitsandfoxes) [inline .github/workflows/unit_tests.yml:75]: I assume this step fails if the tests fail?

Comment (limbonaut) [reply .github/workflows/unit_tests.yml:75]: Strangely, docs don't specify. I'll test.

Comment (limbonaut) [reply .github/workflows/unit_tests.yml:75]: ~~I'm not sure about `timeout-minutes` - docs don't specify if it fails the step. Need to check it.~~ (Works as expected).
According to docs, the process should return:
```
0   = all tests success
100 = ends with test failures
101 = ends with test warnings
```

**Pull request merged**